leftwm-core/src/handlers/command_handler.rs

```diff
@@ -375,6 +377,44 @@ fn focus_window(state: &mut State, param: &str) -> Option<bool> {
     }
 }
 
+// TODO: add comment
+fn focus_window_direction(state: &mut State, dir: FocusDirection) -> Option<bool> {
+    let workspace = state.focus_manager.workspace(&state.workspaces)?.rect();
+    let mut rects: Vec<Rect> = vec![];
+    let cur_window = state.focus_manager.window(&state.windows)?;
+
+    let mut cur = 0;
```

Comment:
I would prefer an Option here, as you're otherwise asserting the focused window is always on the focused workspace, which leftwm does not guarantee (edge-cases with Sloppy focus and DisplayEvent/DisplayAction buffers may apply).

Reply:
Hi @Eskaan  I was not aware of that. I will make this change right now. Thanks for your time.

Reply:
I did take a closer look at `handle_window_focus` in focus_handler.rs and the edge-case I thought of does not exist, as we are only calling the display server to move the mouse over the window, which then causes xorg to emit a EnterNotify event, which works it's way back to `focus_window`. My mistake was that I assumed `MoveMouseOver` also focused the window before the workspace.

Reply:
Also I was refering to line 388 `let mut cur = 0;`, were I would prefer `let mut cur = None` and replacing line 392 with `cur = Some(i);`. You can then unwrap or return in line 402 with a question mark and return if no window was found: `find_neighbor(&rects, cur?, dir, &workspace)`

Reply:
Hi @Eskaan thanks for your time and for the clarification.
I have added your suggestions with another commit.